src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using Desktop.Analyzers.Common;
+
+
+namespace Desktop.Analyzers
+{
+    public abstract class DoNotUseInsecureCryptographicAlgorithmsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DoNotUseMD5RuleId = "CA5350";
```

Comment:
I'm not sure the distinctions around algorithms are very helpful. We only require granularity in rule id for cases where there might be some mingling of issues that users want to keep separate (suppressing some, not suppressing others). So, if you applied a single exclusion to a method with a common ID, and you happened to intermix two cryptographically weak algorithms in that method, you would suppress violations for both. I don't think this intermixing is common. With Roslyn, you always have the option of using a #pragma to isolate issues (unless this check requires wrapping an entire type).

Long story short, I don't find the argument to create per algorithm ids compelling. Another thing to think about is the evolution of this check. New crypto algorithms may be added to the list (you know, when quantum computers crack all our existing <g>), it'd be nice not to have to author new rules. Any other thoughts out there?